### PR TITLE
feature: added signup page

### PR DESCRIPTION
**Pull request by bottercode**
Added signup Page
- created signup page
- confirmation logic is yet to be implemented
- feedback is yet to be implemented

### CODE REVIEW

**Comment by bottercode** [comment]
@manavsiddharthgupta

File: app/signup/page.tsx [hunk 1]
```diff
@@ -1,9 +1,63 @@
...
+
+  const {inputValue: password, onChangeHandler: onPasswordChangeHandler}= 
+    useInputField((value) => {
+      const passwordRegex = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
+      return passwordRegex.test(value);
+    })
+
+  const {inputValue: confirmPassword, onChangeHandler: onConfirmPasswordChangeHandler}= 
+    useInputField((value) => {
+      const passwordRegex = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
```
Comment (manavsiddharthgupta) [inline app/signup/page.tsx:21]: passwordRegex doesn't matter in confirm password. confirm password should depend on password. password === confirm password.

File: app/signup/page.tsx [hunk 2]
```diff
@@ -1,9 +1,63 @@
...
+      return passwordRegex.test(value);
+    })
+  
   return (
     <main className="min-h-screen flex items-center justify-center">
       <section>
         <p>SignIn</p>
```
Comment (manavsiddharthgupta) [inline app/signup/page.tsx:28]: Its a Sign Up Page

File: app/signup/page.tsx [hunk 3]
```diff
@@ -1,9 +1,63 @@
...
+            value={avatarName}
+            type="text"
+            placeholder="Enter your Avatar Name"
+            label="Avatar Name"
+            onChangeHandler={onAvatarNameChangeHandler}
+            />
+          
+          <InputField
+            value={password}
+            type="text"
```
Comment (manavsiddharthgupta) [inline app/signup/page.tsx:45]: type shouldn't be text for password fields

File: app/signup/page.tsx [hunk 4]
```diff
@@ -1,9 +1,63 @@
...
+            placeholder="Enter your password"
+            label="Password"
+            onChangeHandler={onPasswordChangeHandler}
+            />
+
+          <InputField
+            value={confirmPassword}
+            type="text"
+            placeholder="Enter your password again"
+            label="Password"
```
Comment (manavsiddharthgupta) [inline app/signup/page.tsx:55]: label Shouldn't be password for confirm password

**Review by manavsiddharthgupta — APPROVED** [review]

**Pull request merged**